arangod/Agency/Node.h

```diff
@@ -335,16 +212,10 @@ class Node final {
 
   /// @brief accessor to Node's Array
   /// @return  returns nullopt if not found or type doesn't match
-  std::optional<velocypack::Slice> hasAsArray(std::string const&) const;
-
-  // These two operator() functions could be "protected" once
-  //  unit tests updated.
-  //
-  /// @brief Get node specified by path string
-  Node& getOrCreate(std::string_view path);
+  Array const* hasAsArray(std::string const&) const;
 
   /// @brief Get node specified by path string
-  Node const* get(std::string_view path) const;
+  NodePtr get(std::string_view path) const;
```

Comment:
Let's move this method up to the other `get` methods and add a comment here that the `getTYPE` methods check that the current node has a certain type and return it, or `None` if the type does not match. This gives then some clartity for the user about the API.